lib/post_action_creator.rb

```diff
@@ -37,6 +37,18 @@ def create(
         create(created_by, post, action, message: message)
       end
     end
+
+    def notify_user_handlers
+      @notify_user_handlers ||= Set.new
+    end
+
+    def add_notify_user_handler(&block)
+      notify_user_handlers << block
+    end
+
+    def reset_notify_user_handlers
+      @notify_user_handlers = Set.new
+    end
```

Comment:
I think it would be better to have this in the DiscoursePluginRegistry along with other plugin-registered callbacks.  That also provides convenience functions for checking whether a plugin is enabled.

We can then add an `add_notify_user_handler` method to `plugin/instance.rb` for convenience

e.g. https://github.com/discourse/discourse/blob/c6aad63a371ea0ecbbd833503e0278d1234f58f1/lib/discourse_plugin_registry.rb#L110-L110

Reply:
being lazy, yes this is better. and honestly not more code.